go/src/koding/kites/kloud/stack/provider/apply.go

```diff
@@ -365,12 +365,14 @@ func (bs *BaseStack) UpdateResources(state *terraform.State) error {
 		return err
 	}
 
+	var merr error
```

Comment:
You don't need this variable, it already works like that. With the exception of L383 - rename:

```go
if cred, err := bs.Builder.CredentialByProvider(machine.Provider); err == nil
```

to

```go
if cred, e := bs.Builder.CredentialByProvider(machine.Provider); e == nil {
```

Also improve error message to sth like:

```go
err = multierror.Append(err, fmt.Errorf("machine %q: no credential found for %q provider: %s", label, machine.Provider, e))
```